### ISSUE TITLE
Implement the Video List API and Visualize Data

### ISSUE THREAD

**Issue by map34** [ADLSOceanHub/OceanHub#14]
## Introduction

The purpose of this REST API is to retrieve a list of videos that are stored inside the [Ocean Observatories website](https://rawdata.oceanobservatories.org/files). From this API, we should be able to retrieve an array of video metadata. There are two parts of this project: implementing the GET endpoint and use the information from the API to display in the frontend.

Concepts gained from here:

- REST API
- Package management
- Python best practices
- Unit tests

### Endpoint Specification

The endpoint that we should implement is

`GET /videos/`

Hitting this endpoint should return:
```javascript
[
    {
        "id" [String],
        "name": [String],
        "frame_count": [Integer],
        "date": [String],
        "url": [String]
    },
    .
    .
    .
]
```

The descriptions of each property of the metadata is as follows:
- "id": This should represent a unique representation of each video, since we do not know the ID of the videos retrieved, let us make one by hashing the URL (using md5). This will be used both internally in our Kafka system, and the endpoint.

- "name": This should be just a human readable name format, could be the last part of the URL for now.

- "frame_count": This should return the number of frames the video has.

- "date": This should be the date of when the video was taken. This should conform to the ISO8601 format.

- "url": Return the url of the video.

### Steps Overview

1. Play around with either [Beautiful Soup](https://www.crummy.com/software/BeautifulSoup/) or [PyCamHD](https://github.com/tjcrone/pycamhd) on a script somewhere. The PyCamHD library especially has a function where you can get a list of MOV files in the server. The link to get the files is under `https://rawdata.oceanobservatories.org/files/`. PyCamHD seems to suffice, but if you need to get something from Beautiful Soup, then that is fine.
2. Once you are comfortable with working with the libraries, add them to `backend/requirements.txt` then rebuild the Docker images. Be aware that if you think you don't need both of them, don't add both of them.
3. Add your endpoint under `backend/server/oceanhub/videos/views.py`, read up a little bit on Flask's endpoint setup in [here](http://flask.pocoo.org/docs/0.12/quickstart/#http-methods).
4. Make sure you `jsonify` the response, we want to return a JSON-serialized response.
5. Add tests to the endpoint, use the [How to Test your Flask Application](http://flask.pocoo.org/docs/1.0/testing/) to add the tests. Also, test your class / functions if you create them, for the code level tests, use Py.test, here is a quick intro to it: https://github.com/pluralsight/intro-to-pytest.

## Additional Information

- @scaraclette will be working on this mostly. But, if anybody would like to get involved, feel free to say so then we will try to divide the work.
- Initial soft-deadline is at August 14th, 2018 for Part 1, but if you need more time let us know. This is just to keep track of time.

**Comment by map34** [comment]
@scaraclette any updates here? Thanks!

**Comment by scaraclette** [comment]
Will work on it soon

**Comment by lsetiawan** [comment]
@map34 @scaraclette Ooooh are we starting this up again? :smile:

**Comment by map34** [comment]
@lsetiawan I am actually not against resurrecting this project again 💯

**Comment by lsetiawan** [comment]
@map34 I've started tweaking things to get it working again :)

**Comment by map34** [comment]
@scaraclette @lsetiawan Hello all, what's the status of this project? Should we replace this w/ Rust?

**Comment by lsetiawan** [comment]
I'm down in trying to implement all of this in Rust! Good learning 🙂

**Comment by scaraclette** [comment]
Yeah this is easy in Rust. Literally just a webscraper. I can set something up

**Comment by map34** [comment]
@scaraclette Any updates here?
